### DISCUSSION TITLE
What would you like to see us build next?

### DISCUSSION THREAD

**Discussion by TuringLovesDeathMetal** [lightdash/lightdash#1488] [Ideas]
### We want to hear your ideas 💡 

Tell us what you'd love to see us build at Lightdash!

Your ideas don't need to be polished. We just want to know how you'd finish this sentence:

> Wouldn't it be cool if Lightdash...

Check out what other folks have written below. If you like someone's idea, give it an upvote 😊

**Comment by TuringLovesDeathMetal** [comment]
Wouldn't it be cool if Lightdash let you embed your charts in Notion/Google Slides/Google docs. And when the chart updated, it also updated in the embedded image

**Reply by JohnRomanski** [reply]
Those may be two different things, one that Lightdash could provide, and one that requires the integration to do. It would be great if there was a URL for a visualization that was constant (/MAU_6months.svg) that always returned the updated data, enabling a standard publish/subscribe model.

**Comment by loomalaine** [comment]
Wouldn't it be cool if Lightdash let you create a color palette and set fonts so that charts and dashboards could fit with company branding

A small thing but, in some places, critical to adoption!

**Reply by owlas** [reply]
This is really essential!

**Reply by TuringLovesDeathMetal** [reply]
#1572 🙌

**Reply by JohnRomanski** [reply]
Also some simple fonts or spacing would go a long way.

**Comment by rrmave** [comment]
A suggestion would be to have scripting funtionality in a dashboard, like a JS extension. The required corporate colour template could be scripted it.

Something like
var colors = [
    '#4F2170',
    '#E18719',
......
]
dashboard.on('initialized', function(dashboard, ev) {
    activeDashboard.style.setPalette(colors, true);
.....
}

A caveat is that the user setting the template would require knowledge of JS but a solution would be to provide a scripting template. The user would then only need to change the colours as required.

**Comment by deepskydata** [comment]
Wouldn't it be cool, if I could define my charts and dashboards by using config code?

**Reply by TuringLovesDeathMetal** [reply]
oooh! @deepskydata we have a similar-ish issue open here: https://github.com/lightdash/lightdash/issues/369

But it's more about writing content _**into**_ code (rather than writing code, and seeing content). 

But I feel like they go really hand-in-hand. Would love to hear any thoughts you might have on it!

**Reply by rawkode** [reply]
A Pulumi provider for dashboards would be fun 😀

**Comment by JohnRomanski** [comment]
Wouldn't it be cool if Lightdash dashboards had a slideshow interface? (the Demo site seems to be down so I can't see the interface)
- the dashboards have clear 16x9 slides defined (with visual indicators like pages or blocks and snap to grid that didn't allow elements to cross borders)
- there were some simple text formatting block options for titles and for captions on the text.
- there were simple UI elements to scroll/flip the pages from slide to slide (left right arrow)
- there were keyboard controls (f to toggle full screen no chrome, right / left arrow to advance, space bar to advance)
- images, gifs and movies would have to be embeddable but not MVP (Could be done through an HTML block of content)

**Reply by JohnRomanski** [reply]
https://www.mckinsey.com/business-functions/growth-marketing-and-sales/our-insights/how-us-consumers-are-feeling-shopping-and-spending-and-what-it-means-for-companies checkout the slideshow style aspect of this report. This is among the highest fidelity bespoke visualization that money can buy, all of the SVGs hand tuned in Adobe Illustrator based on the source code comments.

But, why not?

**Comment by JohnRomanski** [comment]
Wouldn't it be cool if Lightdash visualizations could be set to be a fixed width/height? liquid layouts are cool, but designers often want to have a predictable and consistent output for print and screen presentation.

**Comment by Martin-Carlsson** [comment]
Custom formats, like Excel custom formats

<img width="617" alt="image" src="https://user-images.githubusercontent.com/7769335/175814556-75c4e74b-4ed6-405e-aae0-9a6d8e8fa1df.png">

**Reply by TuringLovesDeathMetal** [reply]
Nice! Yes, Tableau is our idol for formatting 😍 It's where we're hoping to be (very soon!) when it comes to formatting: having the drop-down, hold-your-hand options, but also having the fully-custom option if you're a pro.

Here are the formatting options we have for now: https://docs.lightdash.com/guides/formatting-your-fields

If there's a really important one missing for you there, please let us know!

Out of interest, @Martin-Carlsson - how often to you use the fully `custom` input box vs. clicking on the existing options they have for each category? (e.g. clicking `scientific`, then clicking through the drop-down options they offer)

**Reply by Martin-Carlsson** [reply]
@TuringLovesDeathMetal 

Nothing **really** important right now. 

Personally I use the custom quite a lot since I'm from Denmark and our currency are written as `100 kr` or `100 kr.` and we have switched `,` and `.` in numbers. So `1.100,50` would be _one thousand one hundred and a half_.

**Comment by JacoBezuidenhout** [comment]
The possibility of pulling in data from REST APIs would be awesome....!

**Comment by ceciliadataedge** [comment]
Wouldn't it be cool if in Lightdash you could change background color of dashboards and charts/visuals. Also changing text size and othe formatting bits in different charts would be cool.

**Comment by timhiebenthal** [comment]
Just played around with Lightdash for the very first time.

Having used mainly Looker Studio, Power BI and Metabase in missed the following features:

- Ability to bookmark charts & dashboards
- Date Picker for filtering dates
- Time Intelligence (compare vs last week, last month, last X days, ...)
- Map dimension values to certain colors (e.g. "cancelled" always has red, "ongoing" always green etc.)
- Define hierarchies between dimensions and allow to drill up/down along this hierarchy (e.g. from month to week to date or from channel to campaign to ad)

**Reply by TuringLovesDeathMetal** [reply]
@timhiebenthal thanks for the feedback!

For this one:

> Date Picker for filtering dates

Could you expand a bit? (we have a date picker for filtering dates when you click on the date field, so I'm just wondering which bit felt broken here). Maybe it's something similar to #3599 ?

**Reply by timhiebenthal** [reply]
Shame on me!
You actually have indeed a date picker.
![image](https://user-images.githubusercontent.com/57394222/205032745-696691fa-b8c7-414d-a89d-3f9db1b1d16d.png)

But indeed I'd expect something more "easy" as in #3599 
It's super common to have fixed time ranges between two dates (and - to my understanding - currently you would need two different filters to replicate it in Lightdash)

I personally love the UX of the Data Studios date filter: 
* If you click on the filter widget, the picker (calendar) automatically pops up 
* but you can also select some default periods 
* or customize your own

![image](https://user-images.githubusercontent.com/57394222/205033456-7179bdc6-6d1c-435b-8f91-d76802d110c0.png)

![image](https://user-images.githubusercontent.com/57394222/205033549-1a7a0d47-3c66-4bb0-8cc1-cd9fa75cfa46.png)

**Reply by TuringLovesDeathMetal** [reply]
great feedback, thanks @timhiebenthal !

CC: @PriPatel here's some more feedback on date pickers 👀

**Reply by PriPatel** [reply]
Nice! I think there is a similar-ish component available to us that we could try using to achieve this:

https://github.com/lightdash/lightdash/issues/3599#issuecomment-1328914262
You can easily select a 'range' without needing to add multiple filters!
